sdjournal/journal_test.go

```diff
@@ -343,3 +343,44 @@ func TestJournalReaderSmallReadBuffer(t *testing.T) {
 		t.Fatalf("Got unexpected message %s", got[1])
 	}
 }
+
+func TestJournalGetUniqueValues(t *testing.T) {
+	j, err := NewJournal()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer j.Close()
+
+	uniqueString := fmt.Sprintf("JOURNALTEST-%d", time.Now().UnixNano())
+	testEntries := []string{"A", "B", "C", "D"}
+	for _, v := range testEntries {
+		if err := journal.Send(uniqueString, journal.PriInfo, map[string]string{"JOURNALTEST": v}); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	time.Sleep(time.Second)
```

Comment:
There are no filters involved here, so I think it should be possible to wait. Perhaps a wait after each send to be sure that all four values are flushed to journal?